sdm-api-app.groovy

```diff
@@ -436,7 +436,7 @@ def processThermostatTraits(device, details) {
     ecoHeatPoint ? sendEvent(device, [name: 'ecoHeatPoint', value: new Double(ecoHeatPoint).round(1)]) : null
     coolPoint ? sendEvent(device, [name: 'coolingSetpoint', value: new Double(coolPoint).round(1)]) : null
     heatPoint ? sendEvent(device, [name: 'heatingSetpoint', value: new Double(heatPoint).round(1)]) : null
-    temp ? sendEvent(device, [name: 'temperature', value: new Double(temp).round(1)]) : null
+    temp ? sendEvent(device, [name: 'temperature', value: new Double(temp).round(1), unit: '°' + getTemperatureScale()]) : null
 }
```

Comment:
we should apply this to the 4 attributes above as well:
* ecoCoolPoint
* ecoHeatPoint
* coolPoint
* heatPoint

Reply:
dunno.  i can tell the "temperature" and "humidity" attribute need units to match the ootb driver behavior.  however, i don't have an ootb device with a cooling point, heating point or eco point though to see what the dashboard thermostat template does and i don't care to create one to test.  to be honest, a unit on the cooling point, heating point or eco point might make the thermostat template jumbled without some additional css to sort it out.  

the reason i did the temperature and humidity attribute is because the ootb temperature and humidity devices are numerous and the units show on those templates and it's easy to see and make consistent.  very rarely if ever will somebody have two thermostats with different drivers. 

i don't use a thermostat template but if i did i don't think i would want the unit on the target points.  this is primarily a dashboard thing.  the unit doesn't show up on the edit device page itself or anywhere else as far as i know (other than event logs).

Reply:
dunno. i can tell the "temperature" and "humidity" attribute need units to match the ootb driver behavior. however, i don't have an ootb device with a cooling point, heating point or eco point though to see what the dashboard thermostat template does and i don't care to create one to test. to be honest, a unit on the cooling point, heating point or eco point might make the thermostat template jumbled without some additional css to sort it out.

the reason i did the temperature and humidity attribute is because the ootb temperature and humidity devices are numerous and the units show on those templates and it's easy to see and make consistent. very rarely if ever will somebody have two thermostats with different drivers.

i don't use a thermostat template but if i did i don't think i would want the unit on the target points. this is primarily a dashboard thing. the unit doesn't show up on the edit device page itself or anywhere else as far as i know (other than event logs).